### ISSUE TITLE
Audit Log file access in nfs-ganesha

### ISSUE THREAD

**Issue by rvelasquezsos** [nfs-ganesha/nfs-ganesha#645]
Hi, I have a CEPH cluster with an exports folders with NFS Ganesha and CephFS as backend.
A user is reporting that someone deleted files in his folder.
Is there any way to audit the access of files in NFS Ganesha? Something like IP address of connection, file accessed, date and time.

Thanks in advance,
RV

**Comment by dang** [comment]
Not really, no.  If you have high enough logging on, you can examine the logs and figure it out, but we don't have auditing support in Ganesha.

**Comment by rvelasquezsos** [comment]
Thanks @dang the problem is when I enable full debug the storage is full in few hours.

**Comment by ffilz** [comment]
Is there any Ceph auditing? That would at least tell you the user assuming the clients are using users and not everyone coming in as root or some generic user.

We could take an enhancement request (re-open and label as an enhancement) to add some audit logging. It might be a low priority, but we could at least track the interest in such logging.

**Comment by rvelasquezsos** [comment]
Hi @ffilz with Ceph we can audit access through Cephx authentication but it would be with "clients" that in this case are gateways accessing the cluster. 
If you can evaluate the option as an enhancement it would be great.

Thanks again

**Comment by ffilz** [comment]
I edited the title to make it clear we are looking at audit logging and marked this as an enhancement.

We will take this under consideration to add as a feature.

We also welcome new contributors to add features they have a need for.
